GVFS/GVFS.Common/LibGit2RepoPool.cs

```diff
@@ -9,24 +9,38 @@ namespace GVFS.Common
     public class LibGit2RepoPool
     {
         private const int TryAddTimeoutMilliseconds = 10;
-        private const int TryTakeTimeoutMilliseconds = Timeout.Infinite;
+        private const int TryTakeWaitForever = Timeout.Infinite;
+        private const int TryTakeWaitShort = 10;
 
         private readonly BlockingCollection<LibGit2Repo> pool;
+        private readonly Func<LibGit2Repo> repoFactory;
         private readonly ITracer tracer;
 
+        private readonly Timer repoDisposalTimer;
+        private readonly TimeSpan repoDisposalDueTime = TimeSpan.FromMinutes(1);
+        private readonly TimeSpan repoDisposalPeriod = TimeSpan.FromSeconds(15);
+        private int numDisposedRepos;

```

Comment:
What is `numDisposedRepos` tracking? I see we increment it when disposing repositories, but also increment it at times as well.

Did you consider keeping the original size of the pool, and the current number of "open" repositories? or maybe naming this `numAvailableRepos` (vs `numActiveRepos`) to capture the idea that this is the number of repository objects that we can spin up?

Reply:
The tricky part is that the size of the collection isn't important, as we remove repos in use from the collection. We can't track the size, but instead track how many we have disposed (which means how many we can reallocate).

Reply:
> which means how many we can reallocate

This is more what I meant. Instead of tracking how many repositories have disposed, should we rather track how many repositories we can still open? This might just be a name change, as you said this already tracks how many we can (re)allocate, but I think it would more accurately reflect the concept it is tracking and is a bit clearer.

As is, I did ask myself if we cared about the number of times we have disposed a repository, or if cared about the net number of disposed repositories vs the number we have re-allocated. 

> The tricky part is that the size of the collection isn't important

Just to be clear - what I meant was the maximum number of open repos, which we specify when creating the Pool (but we do not keep around). It seems like a pool has the following properties (which may be implicit):

1) The maximum number of open repositories
2) The current number of open repositories
3) The number of repositories that we can open

Some of these numbers can be calculated from the others. With this change, you want to track property 3.